django/db/migrations/serializer.py

```diff
@@ -0,0 +1,396 @@
+from django.utils.timezone import utc
+from django.utils.version import get_docs_version
+
+try:
+    import enum
+except ImportError:
+    # No support on Python 2 if enum34 isn't installed.
+    enum = None
+
+
+class BaseSerializer(object):
+    def __init__(self, value):
+        self.value = value
+
+    def serialize(self):
+        raise NotImplementedError
+
+
+class SimpleSerializer(BaseSerializer):
+    def serialize(self):
+        return repr(self.value), set()
+
+
+class FloatSerializer(SimpleSerializer):
+    def serialize(self):
+        if math.isnan(self.value) or math.isinf(self.value):
+            return 'float("{}")'.format(self.value), set()
+        return super(FloatSerializer, self).serialize()
+
+
+class SequenceSerializer(BaseSerializer):
+    def _format(self):
+        return "[%s]"
+
+    def serialize(self):
+        imports = set()
+        strings = []
+        for item in self.value:
+            item_string, item_imports = serializer_factory(item).serialize()
+            imports.update(item_imports)
+            strings.append(item_string)
+        format = self._format()
+        return format % (", ".join(strings)), imports
+
+
+class FrozensetSerializer(SequenceSerializer):
+    def _format(self):
+        return "frozenset([%s])"
+
+
+class SetSerializer(SequenceSerializer):
+    def _format(self):
+        # Don't use the literal "{%s}" as it doesn't support empty set
+        return "set([%s])"
+
+
+class TupleSerializer(SequenceSerializer):
+    def _format(self):
+        # When len(value)==0, the empty tuple should be serialized as
+        # "()", not "(,)" because (,) is invalid Python syntax.
+        return "(%s)" if len(self.value) != 1 else "(%s,)"
+
+
+class DictionarySerializer(BaseSerializer):
+    def serialize(self):
+        imports = set()
+        strings = []
+        for k, v in sorted(self.value.items()):
+            k_string, k_imports = serializer_factory(k).serialize()
+            v_string, v_imports = serializer_factory(v).serialize()
+            imports.update(k_imports)
+            imports.update(v_imports)
+            strings.append((k_string, v_string))
+        return "{%s}" % (", ".join("%s: %s" % (k, v) for k, v in strings)), imports
+
+
+class EnumSerializer(BaseSerializer):
+    def serialize(self):
+        enum_class = self.value.__class__
+        module = enum_class.__module__
+        imports = {"import %s" % module}
+        v_string, v_imports = serializer_factory(self.value.value).serialize()
+        imports.update(v_imports)
+        return "%s.%s(%s)" % (module, enum_class.__name__, v_string), imports
+
+
+class DatetimeSerializer(BaseSerializer):
+    def serialize(self):
+        value_repr = self.serialize_datetime()
+        imports = ["import datetime"]
+        if self.value.tzinfo is not None:
+            imports.append("from django.utils.timezone import utc")
+        return value_repr, set(imports)
+
+    def serialize_datetime(self):
```

Comment:
I actually moved it from MigrationWriter. I guess it doesn't need to be a separate method. I'll amend this shortly.